### ISSUE TITLE
TransformerDataset casting float Y-value to 'tf.int64'

### ISSUE THREAD

**Issue by sagar-spkt** [amaiya/ktrain#129]
https://github.com/amaiya/ktrain/blob/62faa15188c8ebbd4420a59f185c42bee7cbd160/ktrain/text/preprocessor.py#L1196-L1211

![ktrain_bug](https://user-images.githubusercontent.com/22609549/80004295-df795400-84e1-11ea-8195-a3accc0a3aa9.png)

**Comment by amaiya** [comment]
This is intentional.  `ktrain` uses `int64` because [the `transformers` library does this in their own source code](https://github.com/huggingface/transformers/blob/master/src/transformers/data/processors/utils.py).   I vaguely recall some models breaking if int64 is not used, but I don't remember for sure.  Thanks.

**Comment by sagar-spkt** [comment]
Doesn’t it create problem in regression? Look at the target value in the screenshot. Now its all zero. The ground truth for my case lies between 0 and 1. Casting to tf.int64 makes all target zero.

**Comment by amaiya** [comment]
Ok - I understand what you're asking now and now see it in your screenshot.  I looked at the `transformers` library and it looks like they are using `int64` even for regression problems for TensorFlow models.  But, they use a `torch.float` for PyTorch regression.  

Let me look into this further.  

In the mean time, as a workaround,  you can perhaps try transforming your y-values to be, say, between 0 and 1000.   

Thanks for opening issue.

**Comment by amaiya** [comment]
I have changed the data type of the dependent variable in text regression problems to `tf.float32`.  I have tested it with DistilBert text regression and everything worked..  The change will be included in the next release (v0.14.3).

Thanks again for bringing this to our attention.

```python
for i, (x,y) in enumerate(valtf):
    print(y)
```

```
tf.Tensor([50.], shape=(1,), dtype=float32)
tf.Tensor([20.], shape=(1,), dtype=float32)
tf.Tensor([17.], shape=(1,), dtype=float32)
tf.Tensor([11.], shape=(1,), dtype=float32)
tf.Tensor([15.], shape=(1,), dtype=float32)
tf.Tensor([100.], shape=(1,), dtype=float32)
tf.Tensor([25.], shape=(1,), dtype=float32)

```

**Comment by amaiya** [comment]
Fix is included in 0.14.3 (just released). Thanks for taking time to open issue.
